libpromises/evalfunction.c

```diff
@@ -8451,6 +8452,231 @@ static FnCallResult FnCallNetworkConnections(EvalContext *ctx, ARG_UNUSED const
 
 /*********************************************************************/
 
+struct IsReadableThreadData
+{
+    pthread_t thread;
+    pthread_cond_t cond;
+    pthread_mutex_t mutex;
+    const char *path;
+    FnCallResult result;
+};
+
+static void *IsReadableThreadRoutine(void *data)
+{
+    assert(data != NULL);
+
+    struct IsReadableThreadData *const thread_data = data;
+
+    int ret = pthread_mutex_lock(&thread_data->mutex);
+    if (ret != 0)
+    {
+        Log(LOG_LEVEL_ERR, "Failed to lock mutex: %s",
+            GetErrorStrFromCode(ret));
+        return NULL;
```

Comment:
This should never happen, right? I'm almost inclined to do something like `ProgrammingError()` here.